library/kani/kani_lib.c

```diff
@@ -1,2 +1,37 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#include <assert.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+
+// This is a C implementation of the __rust_realloc function that has the following signature
+//     fn __rust_realloc(ptr: *mut u8, old_size: usize, align: usize, new_size: usize) -> *mut u8;
+// The implementation pretty much follows CBMC's implementation:
+// https://github.com/diffblue/cbmc/blob/94b81b22799876b52632ba9cdc3f486f87abadb8/src/ansi-c/library/stdlib.c#L472
```

Comment:
Can you please clarify in the comments why we need our own?

Reply:
Done